Scripts/aws_stack_provision.ps1

```diff
@@ -1,103 +1,302 @@
+
+      			Start-Sleep -Seconds 120
+      			$RetryCount -= 1
+      			if ( ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$MYSQL_DB_IDENTIFIER}).DBInstanceStatus -eq "starting") -or ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$MYSQL_DB_IDENTIFIER}).DBInstanceStatus -eq "backing-up")  ) {
+
+         			Write-Host "MYSQL RDS is getting started"}}
+
+    		if ( $RetryCount -le 0 ) {
+       			throw "Timeout: 30 minutes expired waiting for MYSQL RDS to be in availabe state" }
+    		Write-Host "MYSQL RDS with Lansa Version tag = $lansa_version is in Available state"}
+
+
+  	else{
+    		throw "MYSQL RDS in $MYSQL_DB_STATUS state"}}
 
 
 elseif ($MYSQL_DB_COUNT -eq 0){
-   echo "No existing MYSQL database exist with required lansa version"
-   $MYSQL_SNAPSHOT_COUNT = ((Get-RDSDBSnapshot -DBInstanceIdentifier $MYSQL_DB_IDENTIFIER).DBSnapshotArn).count
-   if ($MYSQL_SNAPSHOT_COUNT -eq 1){
-     echo "Found 1 snapshot for database identifier $MYSQL_DB_IDENTIFIER"}
-   elseif ($MYSQL_SNAPSHOT_COUNT -eq 0){
-     echo "Found 0 snapshot for database identifier $MYSQL_DB_IDENTIFIER"}
-   else{
-     echo "Found more than 1 snapshot for database identifier $MYSQL_DB_IDENTIFIER"}}
+   	Write-Host "No existing MYSQL RDS with Lansa version tag = $lansa_version exist"
+   	$MYSQL_SNAPSHOT_COUNT = ((Get-RDSDBSnapshot -DBInstanceIdentifier $MYSQL_DB_IDENTIFIER).DBSnapshotArn).count
+   	if ($MYSQL_SNAPSHOT_COUNT -eq 1){
+     		Write-Host "Found 1 snapshot for database identifier $MYSQL_DB_IDENTIFIER"
+
+		## Creating new MYSQL RDS resource
+     		#New-CNFStack -StackName .........................
+
+		## Validating if newly provisioned MYSQL RDS is in available state
+     		$RetryCount = 15
+     		while ( ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$MYSQL_DB_IDENTIFIER}).DBInstanceStatus -ne "available") -and ($RetryCount -gt 0)){
+
+       			Start-Sleep -Seconds 120
+       			$RetryCount -= 1 
+       			if ( ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$MYSQL_DB_IDENTIFIER}).DBInstanceStatus -eq "starting") -or ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$MYSQL_DB_IDENTIFIER}).DBInstanceStatus -eq "backing-up")  ) {
+
+          			Write-Host "RDS is getting started"}}
+
+       		if ( $RetryCount -le 0 ) {
+          		throw "Timeout: 30 minutes expired waiting for MYSQL RDS to be in availabe state" }}
+
+   	elseif ($MYSQL_SNAPSHOT_COUNT -eq 0){
+     		throw "Found 0 snapshot for database identifier $MYSQL_DB_IDENTIFIER"}
+   	else{
+     		throw "Found more than 1 snapshot for database identifier $MYSQL_DB_IDENTIFIER"}}
 
 else{
-   echo "More than 1 MYSQL DB with required lansa version tag already exist"}
+   	throw "More than 1 MYSQL RDS with lansa version tag = $lansa_version already exist"}
+
+############################################################################################################################################################
+
```

Comment:
So with the small additions to use the clone version when looking for the AMI to create the VM from, and restoring the clone snapshots always, this following code is not required.

Reply:
@robe070 
So you want to use the same block of code that we used for normal validation and provision to be used for the clone process as well?

Reply:
Correct. Its doing ALL the same things except about where it gets its saved state from. That is the only difference.

Reply:
And that difference is covered by setting $clone_lansa_version appropriately